File: python/docs_code_snippets/knowledge-delete-documents.py
import time
from opperai import Opper

opper = Opper()

# Setup
kb = opper.knowledge.create(name=f"docs-snippet-{int(time.time())}")
opper.knowledge.add(
    kb.id,
    content="Some outdated content.",
    metadata={"source": "outdated"},
)

# --- docs ---
# Delete documents matching filters
opper.knowledge.delete_documents(
    kb.id,
    filters=[{"field": "source", "operation": "=", "value": "outdated"}],
)
# --- /docs ---

# Cleanup
opper.knowledge.delete(kb.id)

File: typescript/docs_code_snippets/knowledge-delete-documents.ts
import { Opper } from "opperai";

const opper = new Opper();

// setup
const kb = await opper.knowledge.create({ name: `docs-snippet-${Date.now()}` });
const kbId = kb.id;
await opper.knowledge.add(kbId, { content: "Old content", metadata: { source: "outdated" } });
// /setup

// --- docs ---
// Delete documents matching filters
await opper.knowledge.deleteDocuments(kbId, {
  filters: [{ field: "source", operation: "=", value: "outdated" }],
});
// --- /docs ---

console.log("Documents deleted");
// cleanup
await opper.knowledge.delete(kbId);
